### DISCUSSION TITLE
grate support - Extension or Tool in Common?

### DISCUSSION THREAD

**Discussion by franhoey** [cake-build/cake#4149] [Extension Q&A] [answered]
I've been using RoundhousE with Cake for years, but recently discovered  @erikbra is re-writing it as [grate](https://erikbra.github.io/grate/).

> The goal of grate is to be largely backwards compatible with RoundhousE, which is an amazing tool. It is initiated by the main maintainer of RoundhousE for the last three years, please see this issue in the RoundhousE repo for details: https://github.com/chucknorris/roundhouse/issues/438.

I would like to bring grate to Cake and as it's backwards compatible with RoundhousE I thought that would be an easy copy, paste and tweak, but Roundhouse lives in the namespace Cake.Common.Tools.Roundhouse, and I felt this would be better implemented as an Extension.

What are your thoughts, and if running it as an Extension, is there a good one to use as reference? I was thinking the terraform extension might be a good reference?

**Comment by pascalberger** [comment, marked as answer]
Makes sense to implement it as an addin. For a good example it's probably best to checkout one of the recently updated addins which contain a tool wrapper, e.g. [Cake.ESLint](https://github.com/cake-contrib/Cake.ESLint).

Please checkout also [Best practices for writing addins](https://cakebuild.net/docs/extending/addins/best-practices) for addin authoring guidelines.

You might also consider to have the addin hosted in the [Cake Contributions organization](https://cakebuild.net/community/cake-contrib).

**Reply by franhoey** [reply]
Thank you, will do
